integration/convert_test.go

```diff
@@ -219,20 +237,109 @@ func TestConvertAndPush(t *testing.T) {
 	registryConfig := newRegistryConfig()
 	sh, done := newShellWithRegistry(t, registryConfig)
 	defer done()
-	for _, imageName := range convertImages {
-		t.Run(imageName, func(t *testing.T) {
+
+	imageName := nginxImage
+
+	tests := []struct {
+		name                 string
+		pullArgs             []string
+		convert              []string
+		pushArgs             []string
+		expectedConvertError string
+		expectedPushError    string
+	}{
+		{
```

Comment:
This is effectively testing all the permutations of converting 3 images yeah? Wonder if we can make it a little more modular with a loop.

```
	var testMatrix [3]string
	mode := []string{"all", "one"}

	for _, i := range mode {
		for _, j := range mode {
			for _, k := range mode {
				temp[0] = i
				temp[1] = j
				temp[2] = k
				t.Run // ...
			}
		}
	}
```

Something along these lines, and then maybe push this out to a helper function. It's also pretty ugly the thought of changing these test cases at all feels a little daunting in its current state.